libsolidity/ast/ASTUtils.cpp

```diff
@@ -17,13 +17,45 @@
 // SPDX-License-Identifier: GPL-3.0
 
 #include <libsolidity/ast/AST.h>
+#include <libsolidity/ast/ASTVisitor.h>
 #include <libsolidity/ast/ASTUtils.h>
 
 #include <libsolutil/Algorithms.h>
 
 namespace solidity::frontend
 {
 
+namespace
+{
+
+class ASTNodeLocator: public ASTConstVisitor
+{
+public:
+	bool visitNode(ASTNode const& _node) override
+	{
+		if (_node.location().containsOffset(m_pos))
+		{
+			m_closestMatch = &_node;
+			return true;
+		}
+		return false;
+	}
+
+	explicit ASTNodeLocator(int _pos): m_pos{_pos}, m_closestMatch{nullptr} {}
+
+	int const m_pos;
```

Comment:
```suggestion
	int const m_positionInFile;
```

Reply:
@cameel i'm not sure if names like this is a little bit over the top for such a small struct. If at all, I'd prefer an actual file offset type over var names of type int for file offsets (e.g. `cpp struct FileOffset { int value; };`. I like to box primitive types in other projects for better readability / code quality / less accidental mix-ups.

Reply:
A dedicated type would be helpful but I think that good, informative names are still needed. It's not a big difference in a small struct but even then I don't think it's worth saving a few keystrokes. With `positionInFile` I immediately understand it when I see it and can move on quickly while with `pos` I have to stop for a moment and look for context clues. Especially given that you see this struct before you encounter the function that uses it so you do not have that context yet.

BTW, the coding style says not to use abbreviations for names (and I think it's good rule).